packages/core-backend/src/BackendWebSocketService.ts

```diff
@@ -1300,14 +1349,48 @@ export class BackendWebSocketService {
   // =============================================================================
 
   /**
-   * Schedules a reconnection attempt with exponential backoff
+   * Schedules a connection attempt with exponential backoff and jitter
+   *
+   * This method is used for automatic reconnection with backoff:
+   * - Prevents duplicate reconnection timers (idempotent)
+   * - Applies exponential backoff based on previous failures
+   * - Adds jitter (±25%) to prevent thundering herd problem
+   * - Used ONLY for automatic retries, not user-initiated actions
+   *
+   * Call this from:
+   * - connect() catch block (on connection failure)
+   * - #handleClose() (on unexpected disconnect)
+   *
+   * For user-initiated actions (sign in, unlock), call connect() directly instead.
+   *
+   * If a reconnect is already scheduled, this is a no-op to prevent:
+   * - Orphaned timers (memory leak)
+   * - Inflated reconnect attempts counter
+   * - Prematurely long delays
    */
-  #scheduleReconnect(): void {
+  #scheduleConnect(): void {
+    // If a reconnect is already scheduled, don't schedule another one
+    if (this.#reconnectTimer) {
+      return;
+    }
+
+    // Increment attempts BEFORE calculating delay so backoff grows properly
     this.#reconnectAttempts += 1;
 
+    // Calculate delay with exponential backoff
```

Comment:
Nit: Probably not important now, but if we like we might want to reuse the same exponential backoff formula we are using for network requests. The Cockatiel library provides a way to generate a series of backoffs here: https://github.com/connor4312/cockatiel/blob/f87137b9bb55e0cdfb02e7d5c401e47c6f5fcc9f/src/backoff/ExponentialBackoff.ts ([this](https://github.com/connor4312/cockatiel/blob/f87137b9bb55e0cdfb02e7d5c401e47c6f5fcc9f/src/backoff/ExponentialBackoffGenerators.ts#L55) is the formula itself).